pkg/commands/process/settings/rules/ruby/third_parties/google_dataflow.yml

```diff
@@ -0,0 +1,140 @@
+      - variable: DATA_TYPE
+        detection: datatype
+languages:
+  - ruby
+auxiliary:
+  - id: google_dataflow_client_init
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::$<_>::Client.new
+      - pattern: |
+          Google::Cloud::Dataflow.$<METHOD>
+        filters:
+          - variable: METHOD
+            values:
+              - messages
+              - templates_service
+              - flex_templates_service
+              - jobs
+              - metrics
+              - snapshots
+  - id: google_dataflow_config
+    patterns:
+      - pattern: |
+          $<VAR>.configure
+        filters:
+          - variable: VAR
+            detection: google_dataflow_client_init
+      - pattern: |
+          $<VAR>.configure { |$<!>$<_:identifier>| }
+        filters:
+          - variable: VAR
+            detection: google_dataflow_client_init
+  - id: google_dataflow_message_text_classes
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::StructuredMessage.new
+      - |
+        Google::Cloud::Dataflow::$<_>::JobMessage.new
+  - id: google_dataflow_snapshots_client_init
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::Snapshots::Client.new
+      - |
+        Google::Cloud::Dataflow.snapshots
+  - id: google_dataflow_description_classes
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::SnapshotJobRequest.new
+      - pattern: |
+          $<VAR>.get_snapshot
+        fitlers:
+          - variable: VAR
+            detection: google_dataflow_snapshots_client_init
+  - id: google_dataflow_value_classes
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::JobMessage.new
+      - |
+        Google::Cloud::Dataflow::$<_>::CreateJobFromTemplateRequest::ParametersEntry.new
+      - |
+        Google::Cloud::Dataflow::$<_>::ParameterMetadata::CustomMetadataEntry.new
+      - |
+        Google::Cloud::Dataflow::$<_>::StructuredMessage::Parameter.new
+  - id: google_dataflow_templates_service_client_init
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::TemplatesService::Client.new
+      - |
+        Google::Cloud::Dataflow.templates_service
+  - id: google_dataflow_template_metadata_init
+    patterns:
+      - |
+        Google::Cloud::Dataflow::$<_>::TemplateMetadata.new
+trigger: local
+severity:
+  default: low
+  PII: critical
+  PHI: medium
+  PD: high
+skip_data_types:
+  - "Unique Identifier"
+metadata:
+  description: "Do not send sensitive data to Google Dataflow."
+  remediation_message: |
+    ## Description
+    Leaking sensitive data to third-party service is a common cause of data leaks and can lead to data breaches. This rule looks for instances of sensitive data sent to Google Dataflow.
+
+    ## Remediations
+
+    When sending data to third-party services, ensure all sensitive data is removed.
+
+    ## Resources
+    - [Google Dataflow Docs](https://cloud.google.com/ruby/docs/reference/google-cloud-dataflow/latest)
```

Comment:
Ping @markmichon for review 🙏 
From [Google Dataflow docs](https://www.rubydoc.info/gems/google-cloud-dataflow/0.2.0), it's "a managed service for executing a wide variety of data processing patterns", whatever that means, so I've kept the description here vague.